homeassistant/components/homekit/accessories.py

```diff
@@ -162,7 +162,7 @@ async def run_handler(self):
         Run inside the Home Assistant event loop.
         """
         state = self.hass.states.get(self.entity_id)
-        self.hass.async_add_executor_job(self.update_state_callback, None, None, state)
+        self.hass.async_add_job(self.update_state_callback, None, None, state)
```

Comment:
@bdraco found this bug -> passing a callback to the executor. Does this have to run in parallel or could the method just be called too?

Reply:
Maybe make `update_state_callback` a coroutine instead? It needs to schedule work on the executor. Maybe it would be good to be able to await that work?

Reply:
`async_track_state_change(self.hass, self.entity_id, self.update_state_callback)`

@MartinHjelmare Would that be safe to do since its being called from `async_track_state_change` as well?

Reply:
Yes. It would be run with `hass.async_run_job`.
https://github.com/home-assistant/core/blob/d43617c41d6507f2d2b77aadf4fa1ebaf0058b14/homeassistant/core.py#L370-L388

Reply:
Adjusted here https://github.com/home-assistant/core/pull/34560/commits/a04a8fa4b10cf961fe97a88bea0d2f711cfabe14

I'll cherry pick it out and make another PR after some more testing